.travis.yml

```diff
@@ -137,64 +161,79 @@ install:
-    echo 'packages: "optics-sop-*/*.cabal"' >> cabal.project
-    echo 'packages: "optics-th-*/*.cabal"' >> cabal.project
-    echo 'packages: "optics-vl-*/*.cabal"' >> cabal.project
-    echo 'packages: "generic-optics-*/*.cabal"' >> cabal.project
-    echo 'packages: "template-haskell-optics-*/*.cabal"' >> cabal.project
-    echo 'packages: "metametapost-*/*.cabal"' >> cabal.project
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_optics}" >> cabal.project ; fi
+    echo "packages: ${PKGDIR_optics_core}" >> cabal.project
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_optics_extra}" >> cabal.project ; fi
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_optics_sop}" >> cabal.project ; fi
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_optics_th}" >> cabal.project ; fi
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_optics_vl}" >> cabal.project ; fi
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_generic_optics}" >> cabal.project ; fi
+    if ! $GHCJS && [ $HCNUMVER -lt 80800 ] ; then echo "packages: ${PKGDIR_template_haskell_optics}" >> cabal.project ; fi
+    if ! $GHCJS ; then echo "packages: ${PKGDIR_metametapost}" >> cabal.project ; fi
   - |
-    echo "constraints:                 aeson==1.4.3.0"       >> cabal.project
-    echo "constraints:                 statistics==0.15.0.0" >> cabal.project
-    echo "write-ghc-environment-files: always"               >> cabal.project
+    echo "constraints: statistics==0.15.0.0" >> cabal.project
   - "for pkg in $($HCPKG list --simple-output); do echo $pkg | sed 's/-[^-]*$//' | (grep -vE -- '^(generic-optics|metametapost|optics|optics-core|optics-extra|optics-sop|optics-th|optics-vl|template-haskell-optics)$' || true) | sed 's/^/constraints: /' | sed 's/$/ installed/' >> cabal.project.local; done"
   - cat cabal.project || true
   - cat cabal.project.local || true
   # Building...
   # this builds all libraries and executables (without tests/benchmarks)
-  - ${CABAL} v2-build -w ${HC} --disable-tests --disable-benchmarks all | color_cabal_output
+  - ${CABAL} v2-build $WITHCOMPILER --disable-tests --disable-benchmarks all | color_cabal_output
   # Building with tests and benchmarks...
   # build & run tests, build benchmarks
-  - ${CABAL} v2-build -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - ${CABAL} v2-build $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output
   # Testing...
-  - ${CABAL} v2-test -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - if ! $GHCJS ; then ${CABAL} v2-test $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output ; fi
   # Doctest...
-  - "doctest --fast $(find */src  -name '*.hs' ! -path 'template-haskell-optics*/*') -XBangPatterns -XDefaultSignatures -XFlexibleContexts -XFlexibleInstances -XFunctionalDependencies -XDeriveFunctor -XGADTs -XLambdaCase -XMultiParamTypeClasses -XRankNTypes -XScopedTypeVariables -XTupleSections -XTypeFamilies -XTypeOperators -XDataKinds -XTypeApplications -XInstanceSigs"
```

Comment:
Why not test everything in one go?

Reply:
I really don't see a good reason to do it in one go. I doubt it would be noticeably faster, e.g.

I.e. I see it as a very good thing, that we can `doctest` each package in separation.

Reply:
Sounds good. I have no opinion about it anyway, was just curious about the reasoning :+1: